.github/workflows/reusable-azure-ai-translate.yml

```diff
@@ -93,16 +91,55 @@ jobs:
         id: datetime
         run: echo "timestamp=$(date -u '+%B %d, %Y at %H:%M UTC')" >> "$GITHUB_OUTPUT"
 
+      - name: Configure Git identity
+        if: steps.check_pr.outputs.found == 'false'
+        run: |
+          git config --global user.name "github-actions"
+          git config --global user.email "github-actions@github.com"
+
+      - name: Create or reset to branch
+        if: steps.check_pr.outputs.found == 'false'
+        env:
+          BRANCH_NAME: ${{ steps.set_branch_name.outputs.BRANCH }}
+        run: |
+          git checkout -B "$BRANCH_NAME"
+
+      - name: Stage changes
+        if: steps.check_pr.outputs.found == 'false'
+        run: git add .
+
+      - name: Commit changes
+        if: steps.check_pr.outputs.found == 'false'
+        run: |
+          if git diff --cached --quiet; then
+            echo "No changes to commit. Skipping PR creation."
+            exit 0
+          fi
+          git commit -m "Auto-translated PO and JSON files"
+
```

Comment:
When updating a pre-existing pull request, we should also perform a `gh pr update-branch` here to ensure the head branch has the latest changes from the base.

Reply:
Actually, this should always be done _before_ running the translation script to ensure that all new changes since first creating the branch are included. Otherwise, there would never be new translations because the script would just rerun on the same state of the code base.